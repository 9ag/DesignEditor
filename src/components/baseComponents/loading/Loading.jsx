import React from 'react'

class Loading extends React.Component {

  render() {
    const {show} = this.props
    const cn = show ? 'spinner-content' : 'spinner-content d-none'
    return <div class={cn} id="spinnerDemo">
      <div class="loader"></div>
    </div>
  }
  
}

export default Loading