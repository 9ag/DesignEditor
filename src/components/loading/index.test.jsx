
import { shallow, mount, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Loading from './index';


describe("Loading component test", () => {
  
  it('snapshot test', () => {
    const component = renderer.create(<Loading show />).toJSON();
    expect(component).toMatchSnapshot();
  });


});