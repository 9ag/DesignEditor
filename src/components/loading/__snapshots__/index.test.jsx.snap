// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component test snapshot test 1`] = `
<div
  className="wrap"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="spinWrap"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="loader spin"
      style={
        Object {
          "width": 50,
        }
      }
    >
      <svg
        className="circular"
        viewBox="25 25 50 50"
      >
        <circle
          className="path"
          cx="50"
          cy="50"
          fill="none"
          r="20"
          stroke-miterlimit="10"
          stroke-width={3}
        />
      </svg>
    </div>
  </div>
</div>
`;
